### DISCUSSION TITLE
What PBC features are supported?

### DISCUSSION THREAD

**Discussion by leifjacobson** [grimme-lab/xtb#509] [Q&A]
Is there documentation of what PBC features are supported and what features are planned to be supported in the future?  

Thank you for information help you can provide.

**Comment by awvwgk** [comment]
I'm currently working on a library backend in the [`tblite`](https://github.com/awvwgk/tblite) project, which should eventually provide PBC support in `xtb` and replace the currently available implementation. This will most likely be available with version 6.5.0, but I have no concrete time estimate when this will become available.

If you are looking for something more or less complete now, you can use [DFTB+](https://github.com/dftbplus/dftbplus) which already integrates the `tblite` library and has decent support for various PBC features. But you have to compile DFTB+ yourself to get support for periodic xTB calculations. A stable DFTB+ release including xTB support will become available around October/November this year.

To my knowledge, `tblite` in DFTB+ is currently the most complete implementation of xTB with PBC support for both GFN1-xTB and GFN2-xTB.

**Reply by leifjacobson** [reply]
Thank you very much for a quick reply and for the information.  If I am only interested in computing energy (and optionally gradient and stress tensor) would it be best to compile and use the `tblite` library directly from the github repo or to install `DFTB+`?

**Reply by awvwgk** [reply]
`tblite` is only a library backend, unless you want to implement your own simulation package or integrate xTB in an existing framework, you won't get far with this project alone. In case you indeed want to integrate xTB in another project, let me know.

Using DFTB+ with `tblite` gives you a pretty solid atomistic simulation framework for most PBC cases. But keep in mind that the integration in DFTB+ is still work in progress, while I'm pretty confident I got it right, there might still be some bugs. Therefore, feedback is more than welcome.

**Reply by leifjacobson** [reply]
I would plan on writing an ASE calculator for your library if the CLI is easy to use.  It's sounds like you are recommending to use DFTB+ even for this purpose.  Thank you for your advice.  I am happy to file issues if I find any bugs.

**Reply by awvwgk** [reply]
There is an ASE calculator for [`xtb`](https://github.com/grimme-lab/xtb-python/blob/master/xtb/ase/calculator.py) and for [`tblite`](https://github.com/awvwgk/tblite/blob/main/python/tblite/ase.py) already available. Feel free to share improvements to the calculators.

**Reply by leifjacobson** [reply]
great.  I ran into issues using the `xtb` calculator which is what motivated this question.  I'll checkout the `tblite` calculator.
